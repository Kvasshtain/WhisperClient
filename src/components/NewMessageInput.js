import React from 'react'
import PropTypes from 'prop-types'

class NewMessageInput extends React.Component {

    state = {
        inputValue: ''
    }

    onSubmit = () => {
        this.setState({
            inputValue: ''
        })

        this.props.onSubmitNewMessage(this.state.inputValue)
    }

    updateInputValue = (eventArg) => {
        this.setState({
            inputValue: eventArg.target.value
        })
    }

    render() {
        return (
            <div>
                <input type="text" onChange = {this.updateInputValue} name = "userMessage" value = { this.state.inputValue }/>
                <button type="submit" onClick = {this.onSubmit}>Send message</button>
            </div>
        )
    }
}

NewMessageInput.propTypes = {
    onSubmitNewMessage: PropTypes.func.isRequired
}

export { NewMessageInput }