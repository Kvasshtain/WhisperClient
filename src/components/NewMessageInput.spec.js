import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json';

import { NewMessageInput } from './NewMessageInput'

describe('NewMessageInput', () => {
    const mockOnSubmitNewMessage = jest.fn()

    it ('should render correctly', () => {
        const testedComponent = shallow(<NewMessageInput 
            onSubmitNewMessage = {mockOnSubmitNewMessage}
        />)

        expect(shallowToJson(testedComponent)).toMatchSnapshot();
      })

})

describe('when typing new message', () => {
    const mockOnSubmitNewMessage = jest.fn()
    const newTestMessage = 'Some new message'

    const testedComponent = shallow(<NewMessageInput
        onSubmitNewMessage={mockOnSubmitNewMessage}
    />)

    beforeEach(() => {
        testedComponent.find('input').simulate('change', {
            target: {
                value : newTestMessage,
            }
        })
    })

    it('updates inputValue field in state', () => {
        expect(testedComponent.state().inputValue).toEqual(newTestMessage);
    })

})

describe('when push the button', () => {
    const mockOnSubmitNewMessage = jest.fn()

    const myComponent = shallow(<NewMessageInput
        onSubmitNewMessage={mockOnSubmitNewMessage}
    />)

    beforeEach(() => {
        myComponent.find('button').simulate('click', {
            preventDefault: () => { },
        })
    })

    it('calls the props.onSubmit', () => {
        expect(mockOnSubmitNewMessage).toHaveBeenCalledTimes(1);
    })

})