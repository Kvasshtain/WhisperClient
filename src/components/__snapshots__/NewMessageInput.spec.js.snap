// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMessageInput should render correctly 1`] = `
<div>
  <input
    name="userMessage"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
    type="submit"
  >
    Send message
  </button>
</div>
`;
